altinn-client/src/main/java/no/difi/meldingsutveksling/AltinnWsClient.java

```diff
@@ -54,29 +61,47 @@ public void send(UploadRequest request) {
 
     private void upload(UploadRequest request, String senderReference) {
 
-        BrokerServiceExternalBasicStreamedSF brokerServiceExternalBasicStreamedSF
-                = new BrokerServiceExternalBasicStreamedSF(configuration.getStreamingServiceUrl());
-
-        IBrokerServiceExternalBasicStreamed streamingService = brokerServiceExternalBasicStreamedSF.getBasicHttpBindingIBrokerServiceExternalBasicStreamed();
-
-        BindingProvider bp = (BindingProvider) streamingService;
-        bp.getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, configuration.getStreamingServiceUrl().toString());
-        bp.getRequestContext().put(JAXWSProperties.REQUEST_TIMEOUT, REQUEST_TIMEOUT);
-        bp.getRequestContext().put(JAXWSProperties.CONNECT_TIMEOUT, CONNECT_TIMEOUT);
-
         try {
             StreamedPayloadBasicBE parameters = new StreamedPayloadBasicBE();
+            parameters.setDataStream(getDataHandler(request));
+
+            CompletableFuture<Void> altinnUpload = CompletableFuture.runAsync(
+                    () -> uploadToAltinn(request, senderReference, parameters),
+                    taskExecutor);
+
+            try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1141)